### PR TITLE
Feature/formatters

### PR DESCRIPTION
**Pull request by deshack**
Add formatters for console output.

### CODE REVIEW

File: Lib/Formatters/Styles/OutputFormatter.php [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+      $unset[] = $this->background['unset'];
+    }
+    if (!is_null($this->options)) {
+      foreach ($this->options as $option) {
+        $set[] = $option['set'];
+        $unset[] = $option['unset'];
+      }
+    }
+
+    if (0 === count($set))
```
Comment (wraxen) [inline Lib/Formatters/Styles/OutputFormatter.php]: Why not `empty($set)`?

File: Lib/Formatters/Styles/OutputFormatter.php [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+  /**
+   * Set some specific style option.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $option Option name.
+   *
+   * @throws Freyja\Exceptions\invalidArgument if option name is invalid.
```
Comment (wraxen) [inline Lib/Formatters/Styles/OutputFormatter.php]: `Freyja\Exceptions\InvalidArgumentException`

File: Lib/Formatters/Styles/OutputFormatter.php [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+  /**
+   * Utility for setting foreground and background colors.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @param string $element Element to set color to. Accepts foreground, background.
+   * @param string|null $color Color name.
+   */
```
Comment (wraxen) [inline Lib/Formatters/Styles/OutputFormatter.php:270]: Missing `@throws` tag.

File: Lib/Formatters/Styles/OutputFormatter.php [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    if (!in_array($color, array('foreground', 'background')))
+      throw new InvalidArgumentException('Invalid element for setColor(). Accepted values: "foreground", "background"');
+
+    if (is_null($color)) {
+      $this->$element = null;
+      return;
+    }
+
+    if (!isset(static::${'available_'.$element.'_colors'}[$color]))
+      $this->invalidArgument($element.' color', $color);
```
Comment (wraxen) [inline Lib/Formatters/Styles/OutputFormatter.php:281]: This should be called in a `try` block.

File: Lib/Formatters/Styles/OutputFormatter.php [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+
+  /**
+   * Raise exception for invalid argument.
+   *
+   * @since 1.0.0
+   * @access private
+   *
+   * @param string $element Element that is invalid.
+   * @param string $value Invalid value name.
+   */
```
Comment (wraxen) [inline Lib/Formatters/Styles/OutputFormatter.php:296]: Missing `@throws` tag.

File: Lib/Formatters/Styles/OutputFormatter.php (hunk 4 above)
Comment (deshack) [inline Lib/Formatters/Styles/OutputFormatter.php:281]: Nope, I catch and rethrow the exception in `setForeground()` and `setBackground()`.

**Pull request closed without merging**